app/models/deleted_object.rb

```diff
@@ -22,7 +22,7 @@ class DeletedObject < ApplicationRecord
 
   scope :stale, -> do
     where.has do
-      ((id.in DeletedObject.deleted_owner.select(:id)) | (object_type == Service.name)) \
+      ((id.in DeletedObject.deleted_owner.select(:id)) | (object_type == Contract.name)) \
```

Comment:
Where is this used?

Reply:
https://github.com/3scale/porta/blob/933da0da4b12d1296e4960b83ca60f3f2af07f68/config/jobs.rb#L24
https://github.com/3scale/porta/blob/933da0da4b12d1296e4960b83ca60f3f2af07f68/app/workers/janitor_worker.rb#L10
https://github.com/3scale/porta/blob/933da0da4b12d1296e4960b83ca60f3f2af07f68/app/workers/purge_stale_objects_worker.rb#L7

Reply:
u should probably also read https://github.com/3scale/porta/pull/1826#issuecomment-620476336 to understand this change 😄

Reply:
Back then:
We were storing service 'parents' for ending up deleting its children when we do not need them anymore because the parent has already being destroyed (and not being in the DB does not mean have been destroyed over a week ago, when the event has already being destroyed and all its actions have been executed).
Now:
We do not need to save service and its children anymore, but we still need Application/Cinstance/Contract and its children for https://issues.redhat.com/browse/THREESCALE-4806
So now we only remove the Application and its children after one week since it has been destroyed. But we do not need to wait for the Service to have been destroyed as well to clean up from DeletedObject anymore 😄

Reply:
So `DeletedObject.stale` used to mean "Deleted services whose owners (accounts) have been deleted as well". And now it means "Deleted contracts whose owners (services) have been deleted as well"?

Reply:
@guicassolato hell no 😂 
stale means "it is not needed anymore and can be destroyed", it doesn't matter why, but when it comes to why, it used to be "Deleted objects from at least one week ago whose owners (services) have been deleted as well OR they are services themselves" and now it is "Deleted objects from at least one week ago whose owners (contracts) have been deleted as well OR they are services themselves"

Reply:
@guicassolato I think that it is more clear if you read the test.
This is the old test:
https://github.com/3scale/porta/blob/933da0da4b12d1296e4960b83ca60f3f2af07f68/test/unit/deleted_object_test.rb#L32-L88

Reply:
> stale means "it is not needed anymore and can be destroyed", it doesn't matter why, but when it comes to why, it used to be "Deleted objects from at least one week ago whose owners (services) have been deleted as well OR they are services themselves" and now it is "Deleted objects from at least one week ago whose owners (contracts) have been deleted as well OR they are contracts themselves"

Right. It's an `OR`, not an `AND`.